server/src/main/java/com/broadcom/lsp/cobol/core/engine/CobolLanguageEngine.java

```diff
@@ -113,6 +119,18 @@ public CobolLanguageEngine(TextPreprocessor preprocessor, DefaultErrorStrategy d
     return new ResultWithErrors<>(visitor.getSemanticContext(), accumulatedErrors);
   }
 
+  @CheckThreadInterruption
+  protected CobolParser getCobolParser(CommonTokenStream tokens) {
```

Comment:
Guice dynamically creates a subclass that applies interceptors by overriding methods. So can't have private method.
Or else we will have to adopt another lib like CGI etc.

Reply:
I think here it may be package-private

Reply:
Please, mark the affected classes with `@SuppressWarnings("WeakerAccess")` not to show the warnings